go-controller/pkg/ovn/pods.go

```diff
@@ -231,6 +231,52 @@ func (oc *Controller) waitForNodeLogicalSwitch(nodeName string) error {
 	return nil
 }
 
+func getPodAddresses(portName string) (net.HardwareAddr, net.IP, bool, error) {
+	podMac, podIP, err := util.GetPortAddresses(portName)
+	if err != nil {
+		return nil, nil, false, err
+	}
+	if podMac == nil || podIP == nil {
+		// wait longer
+		return nil, nil, false, nil
+	}
+	return podMac, podIP, true, nil
+}
+
+func waitForPodAddresses(portName string) (net.HardwareAddr, net.IP, error) {
+	var (
+		podMac net.HardwareAddr
+		podIP  net.IP
+		done   bool
+		err    error
+	)
+
+	// First try to get the pod addresses quickly using exponential backoff,
+	// then after about 2 seconds fall back to polling every second.
```

Comment:
This seems overly complicated... Do we have any idea about what the expected distribution of response times is?

I'd either do a single `ExponentialBackoff` for the entire duration (maybe with a lower `Factor` to keep the steps from getting *too* far apart) or else replace the `ExponentialBackoff` with another `PollImmediate`.

Reply:
@danwinship I did it like this so that we'd have a reasonable-but-not-hammering the NBDB to start with, trying to get the address fairly quickly, but intervals that do this space out a lot when you get higher. eg 3 -> 6 -> 12 -> 24 etc which I think is probably too much interval at the end. We're just waiting for northd to catch up if it's a bit behind.

Reply:
@danwinship so what about doing a PollImmediate() with 50ms timeouts for 200ms, then 1s timeouts?